halo2_proofs/src/poly/evaluator.rs

```diff
@@ -225,6 +241,19 @@ impl<E, F: Field, B: Basis> Evaluator<E, F, B> {
                     }
                     lhs
                 }
+                Ast::DistributePowers(terms, base) => {
+                    let mut acc = vec![F::zero(); ctx.local_chunk_size()];
+
+                    for term in terms.iter() {
+                        let term = recurse(term, ctx);
+                        for (acc, term) in acc.iter_mut().zip(term) {
+                            *acc *= base;
+                            *acc += term;
+                        }
+                    }
+
+                    acc
+                }
```

Comment:
I would slightly prefer this to use `fold`:
```suggestion
                Ast::DistributePowers(terms, base) => terms.iter().fold(
                    B::constant_term(ctx.poly_len, ctx.chunk_size, ctx.chunk_index, F::zero()),
                    |mut acc, term| {
                        let term = recurse(term, ctx);
                        for (acc, term) in acc.iter_mut().zip(term) {
                            *acc *= base;
                            *acc += term;
                        }
                        acc
                    },
                ),
```
This more closely matches the `reduce` we were originally using (except taking into account the fact that we only have an immutable reference to the AST here, so a `fold` is easier to work with).